### PR TITLE
Install-DbaWhoisActive, fixes #1617

### LINKED ISSUE #1617: Install-DbaWhoisActive, add output as done in #1614
feature request from @potatoqualitee

### PR DESCRIPTION
**Pull request by niphlod**
Fixes #1617  

Changes proposed in this pull request:
 - added output
 - tuned warning message that pointed towards a deprecated cmdlet

How to test this code: 
- [ ] use it and see that "Status" is "Installed"
- [ ] use it again and see that "Status" is "Updated"

### CODE REVIEW

**Comment by potatoqualitee** [comment]
Hey niph,
Can you make it prompt less? Even if it's removing prompts alltogether. Also, I encountered an error, which can be seen below, along with the prompts.

image

File: functions/Install-DbaWhoIsActive.ps1 [hunk 1]
```diff
@@ -186,7 +190,19 @@ function Install-DbaWhoIsActive {
...
 				}
-				
+				$baseres = @{
+					ComputerName = $server.NetName
+					InstanceName = $server.ServiceName
+					SqlInstance  = $server.DomainInstanceName
+					Database     = $Database
+					Name         = $scriptname.TrimEnd('.sql')
+				}
+				if('sp_WhoisActive' -in $allprocedures) {
```
Comment (potatoqualitee) [inline functions/Install-DbaWhoIsActive.ps1]: This should be [pscustomobject]@ right out the gate without assigning to $baseres first.

Then what you can do is decide the status prior to creating the object, and then assign that status

so 

````
if('sp_WhoisActive' -in $allprocedures) {
$status = "Updated"
````

Then here:

````
[pscustomobject]@{
ComputerName = $server.NetName
InstanceName = $server.ServiceName
SqlInstance  = $server.DomainInstanceName
Database     = $Database
Name         = $scriptname.TrimEnd('.sql')
Status = $status
}
````

thanks baw!

**Comment by niphlod** [comment]
the fear of the assigning output to a variable hits again :P 

will edit accordingly ASAP.

**Comment by niphlod** [comment]
@potatoqualitee : there's a specific example with confirm:$false for no prompts: should I just nuclearize every previous attempt to make it "GUI-like" which seems to be the first goal of this cmdlet ?

**Comment by potatoqualitee** [comment]
I recall previously that it was just one prompt but now it's crazy amounts (probably to properly handle whatif). Knock that bad boy's ConfirmImpact back to normal with a :

`[CmdletBinding(SupportsShouldProcess)]`

**Comment by niphlod** [comment]
@potatoqualitee : test it now

**Comment by gbargsley** [comment]
I ran this through some testing and got installed, but could not get it to hit the updated status.

image

**Comment by niphlod** [comment]
Umpf. Will triple check tomorrow.

Il 12 lug 2017 9:33 PM, "gbargsley" <notifications@github.com> ha scritto:

**Comment by niphlod** [comment]
so sorry, @gbargsley it should now work fine.

**Comment by potatoqualitee** [comment]
PR Party! 🎊

**Pull request merged**